eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java

```diff
@@ -441,6 +453,48 @@ private void waitForPopulatedCache() {
         }
     }
 
+    @Test
+    public void testAssignmentEventCorrectlyCreated() {
+        Date testStart = new Date();
+        initClient(TEST_HOST, true, false, false, DUMMY_API_KEY);
+        SubjectAttributes subjectAttributes = new SubjectAttributes();
+        subjectAttributes.put("age", EppoValue.valueOf(30));
+        subjectAttributes.put("employer", EppoValue.valueOf("Eppo"));
+        double assignment = EppoClient.getInstance().getDoubleAssignment("numeric_flag", "alice", subjectAttributes, 0.0);
+
+        assertEquals(3.1415926, assignment, 0.0000001);
+
+        ArgumentCaptor<Assignment> assignmentLogCaptor = ArgumentCaptor.forClass(Assignment.class);
```

Comment:
Borrowing lessons learned from Java Bandit Logger tests! 📈

Reply:
This was very cool - Java has some powerful tools.